### PR TITLE
Enhancement: Added more HTTP headers suggestions

### PR DESCRIPTION
**Pull request by PCoder23**
## PR Description

Added 29 more http header suggestions from source [Link](https://www.geeksforgeeks.org/http-headers/)
these are

1. X-DNS-Proxy-Authenticate
2. Proxy-Authorization
3.  WWW-Authenticate
4. Age
5. Clear-Site-Data
6. Pragma
7. Warnings
8. Accept-CH
9. Accept-CH-Lifetime
10. Content-DPR
11. DPR
12. Device-Memory
13. Early-Data
14. Save-Data
15. Viewport-Width
16. Width
17. Last-Modified
18. ETag
19. Vary
20. Keep-Alive
21. Accept-charset
22. Timing-Allow-Origin
23. TK
24. Alt-Svc
25. Large-Allocation
26. Link
27. Server-Timing
28. SourceMap
29. X-DNS-Prefetch-Control

## Related Issues

- Enhances #105 

### Checklist
- [x] I have gone through the [contributing guide](https://github.com/foss42/apidash/blob/main/CONTRIBUTING.md)
- [x] I have run the tests (`flutter test`) and all tests are passing

## Added/updated tests?
_We encourage you to add relevant test cases._

- [x] Yes

### CODE REVIEW

**Comment by PCoder23** [comment]
please review @ashitaprasad

File: lib/utils/header_utils.dart [hunk 1]
```diff
@@ -1,5 +1,9 @@
+
 Map<String, String> headers = {
+  "Age": "It is a response header. It defines the times in seconds of the object that have been in the proxy cache.",
```
Comment (animator) [inline lib/utils/header_utils.dart:3]: `Age` is a response header not request header so it should be removed.

**Comment by animator** [comment]
Some feedback: 
- geeksforgeeks is not a reputed site for a source so make sure you add some official source from next time.

File: lib/utils/header_utils.dart [hunk 2]
```diff
@@ -16,11 +20,19 @@ Map<String, String> headers = {
...
   "Accept-Language":
       "Specifies the preferred natural language and locale for the response.",
+  "Accept-CH":
+      "It is a response-type header. It specify which Client Hints headers client should include in subsequent requests.",
+  "Accept-CH-Lifetime":
+      "It is a response-type header used to specify persistence of Accept-CH header value.",
   "Authorization":
       "Contains credentials for authenticating the client with the server.",
   "Authorization Bearer Token": "Often used for token-based authentication.",
+  "Alt-Svc":
```
Comment (animator) [inline lib/utils/header_utils.dart:30]: response header not request header

File: lib/utils/header_utils.dart [hunk 3]
```diff
@@ -16,11 +20,19 @@ Map<String, String> headers = {
...
+      "It is use to reach the website in an alternate way.",
   "Cache-Control":
       "Provides directives for caching mechanisms in both requests and responses.",
+  "Clear-Site-Data":
```
Comment (animator) [inline lib/utils/header_utils.dart:34]: response header not request header

File: lib/utils/header_utils.dart [hunk 4]
```diff
@@ -40,9 +52,18 @@ Map<String, String> headers = {
       "Controls which documents are allowed to open a new window or access the current window.",
   "Cross-Origin-Resource-Policy":
       "Controls how cross-origin requests for resources are handled.",
+  "Content-DPR":
```
Comment (animator) [inline lib/utils/header_utils.dart:55]: response header not request header

File: lib/utils/header_utils.dart [hunk 5]
```diff
@@ -40,9 +52,18 @@ Map<String, String> headers = {
...
+      "It is a response-type header. It is used to define the ratio between physical pixels over CSS pixels of the selected image response.",
+  "DPR": "It is response-type header, It is used to defines the ratio of the physical pixels over the CSS pixels of the current window of the device.",
```
Comment (animator) [inline lib/utils/header_utils.dart:57]: response header not request header

File: lib/utils/header_utils.dart [hunk 6]
```diff
@@ -57,9 +78,22 @@ Map<String, String> headers = {
       "Used in conjunction with the Range header to conditionally request a partial resource.",
   "If-Unmodified-Since":
       "Used for conditional requests, allows the server to respond based on certain conditions.",
+  "Keep-Alive":
+      "It is a general-type header used to inform that how long a persistent connection should stay open.",
+  "Last-Modified":
+      "The last modified response header is a header sent by the server specifying the date of the last modification of the requested source. This is the formal definition of Last-Modified of HTTP headers.",
```
Comment (animator) [inline lib/utils/header_utils.dart:84]: response header not request header

File: lib/utils/header_utils.dart [hunk 7]
```diff
@@ -57,9 +78,22 @@ Map<String, String> headers = {
...
   "Location":
       "Indicates the URL a client should redirect to for further interaction.",
+  "Large-Allocation":
```
Comment (animator) [inline lib/utils/header_utils.dart:87]: response header not request header

File: lib/utils/header_utils.dart [hunk 8]
```diff
@@ -57,9 +78,22 @@ Map<String, String> headers = {
...
+      "It is a response-type header that informs supported browsers (currently only Firefox) about the needs of a memory that allows them to make sure that the large-allocation succeeds and also start a new process using some unfragmented memory.",
+  "Link":
+      "It is entity-type header used to serializing one or more links in HTTP headers.",
   "Origin": "Specifies the origin of a cross-origin request.",
+  "Proxy-Authenticate": "It is a response header gives access to a resource file by defining an authorization method. It allows the proxy server to transmit the request further by authenticating it.",
```
Comment (animator) [inline lib/utils/header_utils.dart:92]: response header not request header

File: lib/utils/header_utils.dart [hunk 9]
```diff
@@ -57,9 +78,22 @@ Map<String, String> headers = {
...
+  "Proxy-Authorization":
+      "It is a request type of header. This header contains the credentials to authenticate between the user agent and the user-specified server.",
+  "Pragma":
```
Comment (animator) [inline lib/utils/header_utils.dart:95]: Deprecated. Remove.

File: lib/utils/header_utils.dart [hunk 10]
```diff
@@ -68,14 +102,34 @@ Map<String, String> headers = {
       "Specifies how much information the browser should include in the Referer header when navigating to other pages.",
   "Retry-After":
       "Informs the client how long it should wait before making another request after a server has responded with a rate-limiting status code.",
+  "Save-Data":
+      "It is used to reduce the usage of the data on the client side.",
   "Server": "Indicates the software used by the origin server.",
+  "Server-Timing":
```
Comment (animator) [inline lib/utils/header_utils.dart:108]: response header not request header

File: lib/utils/header_utils.dart [hunk 11]
```diff
@@ -68,14 +102,34 @@ Map<String, String> headers = {
...
+      "It is a response-type header. This header is used to communicate between two or more metrics and descriptions for a given request-response cycle from the user agent.",
+  "SourceMap":
```
Comment (animator) [inline lib/utils/header_utils.dart:110]: response header not request header

File: lib/utils/header_utils.dart [hunk 12]
```diff
@@ -68,14 +102,34 @@ Map<String, String> headers = {
...
+      "It is a response-type header used to map original source from the transformed source. For example, the JavaScript resources are transformed to some other source from its original by the browsers at the time of execution.",
   "Strict-Transport-Security":
       "Instructs the browser to always use HTTPS for the given domain.",
+  "Timing-Allow-Origin":
```
Comment (animator) [inline lib/utils/header_utils.dart:114]: response header not request header

File: lib/utils/header_utils.dart [hunk 13]
```diff
@@ -68,14 +102,34 @@ Map<String, String> headers = {
...
+      "It is a response type header. It specify origins that are allowed to see values of attributes retrieved via features of the Resource Timing API.",
+  "TK":
```
Comment (animator) [inline lib/utils/header_utils.dart:116]: response header not request header

File: lib/utils/header_utils.dart [hunk 14]
```diff
@@ -68,14 +102,34 @@ Map<String, String> headers = {
...
+      "It is a response type header, it indicates the tracking status.",
   "TE": "Specifies the transfer encodings that are acceptable to the client.",
   "User-Agent":
       "Identifies the client software and version making the request.",
+  "Vary":
```
Comment (animator) [inline lib/utils/header_utils.dart:121]: response header not request header

File: lib/utils/header_utils.dart [hunk 15]
```diff
@@ -68,14 +102,34 @@ Map<String, String> headers = {
...
+      "It is response-type header. It is used by the server to indicate which headers it used when selecting a representation of a resource in a content negotiation algorithm.",
   "Via":
       "Indicates intermediate proxies or gateways through which the request or response has passed.",
+  "Viewport-Width":
+      "It is used to indicates the layout viewport width in CSS pixels.",
+  "Width":
+      "It is a request-type header. This header is used indicates the desired resource width in physical pixels.",
+  "Warnings":
+      "It is a general type header that is used to inform possible problems to the client.",
+  "WWW-Authenticate":
```
Comment (animator) [inline lib/utils/header_utils.dart:131]: response header not request header

File: lib/utils/header_utils.dart [hunk 16]
```diff
@@ -89,6 +143,8 @@ Map<String, String> headers = {
       "Indicates whether the request was made with JavaScript using XMLHttpRequest.",
   "X-XSS-Protection":
       "Enables or disables the browser's built-in cross-site scripting (XSS) filter.",
+  "X-DNS-Prefetch-Control":
```
Comment (animator) [inline lib/utils/header_utils.dart:146]: response header not request header

File: lib/utils/header_utils.dart [hunk 17]
```diff
@@ -68,14 +102,34 @@ Map<String, String> headers = {
...
+  "TK":
+      "It is a response type header, it indicates the tracking status.",
   "TE": "Specifies the transfer encodings that are acceptable to the client.",
   "User-Agent":
       "Identifies the client software and version making the request.",
+  "Vary":
+      "It is response-type header. It is used by the server to indicate which headers it used when selecting a representation of a resource in a content negotiation algorithm.",
   "Via":
       "Indicates intermediate proxies or gateways through which the request or response has passed.",
+  "Viewport-Width":
```
Comment (animator) [inline lib/utils/header_utils.dart:125]: Deprecated. Remove.

File: lib/utils/header_utils.dart [hunk 18]
```diff
@@ -68,14 +102,34 @@ Map<String, String> headers = {
...
+      "It is used to indicates the layout viewport width in CSS pixels.",
+  "Width":
```
Comment (animator) [inline lib/utils/header_utils.dart:127]: Deprecated. Remove.

File: lib/utils/header_utils.dart [hunk 19]
```diff
@@ -68,14 +102,34 @@ Map<String, String> headers = {
...
+      "It is a request-type header. This header is used indicates the desired resource width in physical pixels.",
+  "Warnings":
```
Comment (animator) [inline lib/utils/header_utils.dart:129]: Deprecated. Remove.

**Comment by PCoder23** [comment]
ok sending a pr with all the corrections , and from a official source in a bit

**Comment by animator** [comment]
sure @PCoder23

File: test/utils/header_utils_test.dart [hunk 20]
```diff
@@ -1,6 +1,36 @@
...
+// Keep-Alive
+// Accept-charset
+// Timing-Allow-Origin
+// TK
+// Alt-Svc
+// Large-Allocation
+// Link
+// Server-Timing
+// SourceMap
+// X-DNS-Prefetch-Control
```
Comment (animator) [inline test/utils/header_utils_test.dart:33]: Also, make sure you remove all these commented out lines.

Comment (PCoder23) [reply test/utils/header_utils_test.dart:33]: ok

**Comment by PCoder23** [comment]
now please review @animator , removed the incorrect ones and added these only in the list

Accept-Charset
Device-Memory
Forwarded
From
Max-Forwards
Proxy-Authorization
Upgrade-Insecure-Requests

File: lib/utils/header_utils.dart [hunk 21]
```diff
@@ -97,4 +113,4 @@ List<String> getHeaderSuggestions(String pattern) {
         (element) => element.toLowerCase().contains(pattern.toLowerCase()),
       )
       .toList();
-}
+}
```
Comment (ashitaprasad) [inline lib/utils/header_utils.dart:116]: Please do not remove the new line from the EOF.

File: test/utils/header_utils_test.dart [hunk 22]
```diff
@@ -138,4 +139,4 @@ void main() {
       expect(getHeaderSuggestions(pattern), expected);
     });
   });
-}
+}
```
Comment (ashitaprasad) [inline test/utils/header_utils_test.dart:142]: Please do not remove the new line from the EOF.

**Comment by ashitaprasad** [comment]
LGTM 🚀

**Pull request merged**